tests/measurements/test_sample.py

```diff
@@ -374,3 +388,25 @@ class DummyOp(Operator):
 
         with pytest.raises(EigvalsUndefinedError, match="Cannot compute samples of"):
             qml.sample(op=DummyOp(0)).process_samples(samples=np.array([[1, 0]]), wire_order=[0])
+
+
+@pytest.mark.jax
+def test_jitting_with_sampling_on_subset_of_wires():
+    """Test case covering bug in Issue #3904.  Sampling should be jit-able
+    when sampling occurs on a subset of wires. The bug was occuring due an improperly
+    set shape method."""
+    import jax
+
```

Comment:
It would be great to introduce variable shots and wires to cover the different edge cases for shots and n_wires